docs/proposals/device-crd.md

```diff
@@ -0,0 +1,457 @@
+| Node                  | The edge node details to which this device belongs to.|
+
+The `Status` contains the following fields
+
+| Field                 | Description                                                                      |
+|-----------------------|----------------------------------------------------------------------------------|
+|Twins                   | A list of device twins for the properties which can be controlled ( the writable properties defined in the device model ). Each device twin attribute is like a control attribute which can be managed from the cloud or the edge. <br /> The cloud configures the `Expected`state of a device property and this configuration update is pushed to the edge node. The mapper sends a command to the device to change this property value as per the expected state . It receives the `Actual` state of the property once the previous operation is complete and sends the actual state to the cloud. Offline device interaction in the edge is possible via twin properties for control/command operations.|
+|State                  | indicates the current device state ( unknown, running, offline  etc.)|
+
+### Device instance sample
+```yaml
+apiVersion: devices.kubeedge.io/v1alpha1
+kind: Device
+metadata:
+  name: sensor-tag01
+  labels:
+    description: 'TI Simplelink SensorTag 2.0 with Bluetooth 4.0'
+    manufacturer: 'Texas Instruments'
+    model: CC2650
+    macAddress: 'B0:B4:48:BD:0F:83'
+spec:
+  deviceModel:
+    apiVersion: devices.kubeedge.io/v1alpha1
+    kind: DeviceModel
+    namespace: foo
+    name: sensor-tag-model
+#  currently all devices are indirect, maybe we can add it later.
+#  connectionType: indirect/direct
+  protocol:
+    name: ble
+  node:
+    comment: 'SensorTag belongs to test-node'
+    id: test-node
+    relation: my-edge-node
+status:
+  state: online
+  twins:
+    - name: temperature-enable
+      actual:
+        metadata:
+          timestamp: '1550049403598'
+          type: string
+        value: OFF
+      expected:
+        metadata:
+          timestamp: '1550049403598'
+          type: string
+        value: OFF
+```
+
+### CRD API Group and Version Details
+The tables below summarize the group , kind and API version details for the CRDs
+
+* DeviceModel
+
+| Field                 | Description             |
+|-----------------------|-------------------------|
+|Group                  | devices.kubeedge.io     |
+|APIVersion             | v1alpha1                |
+|Kind                   | DeviceModel             |
+
+* DeviceInstance
+
+| Field                 | Description             |
+|-----------------------|-------------------------|
+|Group                  | devices.kubeedge.io     |
+|APIVersion             | v1alpha1                |
+|Kind                   | Device                  |
+
+### Device Model and Instance Types Definition
+
+```go
+// DeviceModelSpec defines the desired state of the device model
+type DeviceModelSpec struct {
```

Comment:
It is better to adjust the indent for the fields and the fields' comments of the struct.
https://github.com/kubeedge/kubeedge/blob/f906d97f18048ded697dceb5c8b2d526eb4addba/docs/proposals/device-crd.md#device-model-and-instance-types-definition

Reply:
applied correct indentation. thanks.